.eslintrc.yaml

```diff
@@ -110,6 +110,9 @@ rules:
   }, {
     selector: "ThrowStatement > CallExpression[callee.name=/Error$/]",
     message: "Use new keyword when throwing an Error."
+    }, {
+    selector: "CallExpression[callee.object.name='assert'][callee.property.name='fail'][arguments.length=1]",
```

Comment:
This came out of specific errors in the code base. I'm unaware of anyone ever misusing `assert.fail()` with two arguments in core, and once you are checking for that, the lint message you have to supply gets a lot trickier.

I don't like ramping up lint rules to be more strict to catch hypothetical problems because as you make the rules more and more strict, you end up generating false positives. But if we were to do that, I'd say let's do it in a different pull request. This one is replacing a custom rule with identical functionality using built-in ESLint capabilities. I'd rather not alter the functionality at the same time.

`assert.fail()` is an unfortunate API IMO. As far as I can tell, if you supply a truthy value for the third argument, the first two arguments and the fourth argument are all ignored.  (I'm not looking at the code so maybe there's an edge case in there where that's not true. I doubt it, though.)  And the first two arguments don't make a lot of sense without the fourth argument. 

So....the useful signatures are `assert.fail(value, value, falsy, value)` or `assert.fail(ignored, ignored, message)`.

It's a mess IMO but this lint rule was designed to catch just the one exceedingly common misuse.